File: app/src/main/java/ru/melod1n/schedule/AboutActivity.java
package ru.melod1n.schedule;

import android.content.res.Configuration;
import android.os.Bundle;
import android.widget.TextView;

import androidx.appcompat.widget.Toolbar;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

import butterknife.BindView;
import butterknife.ButterKnife;
import ru.melod1n.schedule.adapter.AboutAdapter;
import ru.melod1n.schedule.common.AppGlobal;
import ru.melod1n.schedule.current.BaseActivity;
import ru.melod1n.schedule.items.AboutItem;

public class AboutActivity extends BaseActivity {

    @BindView(R.id.app_version)
    TextView version;

    @BindView(R.id.app_name)
    TextView name;

    @BindView(R.id.list)
    RecyclerView list;

    @BindView(R.id.toolbar)
    Toolbar toolbar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_about);
        ButterKnife.bind(this);

        toolbar.setNavigationOnClickListener(view -> onBackPressed());

        LinearLayoutManager manager = new LinearLayoutManager(this);

        if (!getResources().getBoolean(R.bool.is_tablet) && getResources().getConfiguration().orientation == Configuration.ORIENTATION_LANDSCAPE) {
            manager.setOrientation(RecyclerView.HORIZONTAL);
        } else {
            manager.setOrientation(RecyclerView.VERTICAL);
        }

        list.setHasFixedSize(true);
        list.setLayoutManager(manager);

        String version_ = getString(R.string.version_about, AppGlobal.app_version_name, AppGlobal.app_version_code);
        version.setText(version_);

        ArrayList<AboutItem> items = new ArrayList<>();
        items.add(new AboutItem(R.drawable.ic_computer, getString(R.string.danil_nikolaev), getString(R.string.main_developer), "https://t.me/melod1n"));
        items.add(new AboutItem(R.drawable.innomax, getString(R.string.max_hubach), getString(R.string.backend_logic_developer), "https://t.me/innomaxx"));

        AboutAdapter adapter = new AboutAdapter(this, items);
        list.setAdapter(adapter);
    }
}

File: app/src/main/java/ru/melod1n/schedule/activity/AboutActivity.kt
package ru.melod1n.schedule.activity

import android.content.res.Configuration
import android.os.Bundle
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import kotlinx.android.synthetic.main.activity_about.*
import kotlinx.android.synthetic.main.recycler_view.*
import ru.melod1n.schedule.R
import ru.melod1n.schedule.adapter.AboutAdapter
import ru.melod1n.schedule.base.BaseActivity
import ru.melod1n.schedule.common.AppGlobal
import ru.melod1n.schedule.api.model.About

class AboutActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_about)

        prepareToolbar()
        prepareRecyclerView()
        prepareAppVersion()

        createAdapter(createItems())
    }

    private fun createItems(): ArrayList<About> {
        return arrayListOf(
                About(R.drawable.ic_computer, getString(R.string.danil_nikolaev), getString(R.string.main_developer), "https://t.me/melod1n"),
                About(R.drawable.innomax, getString(R.string.max_hubach), getString(R.string.backend_logic_developer), "https://t.me/innomaxx")
        )
    }

    private fun createAdapter(items: ArrayList<About>) {
        val adapter = AboutAdapter(this, items)
        recyclerView.adapter = adapter
    }

    private fun prepareAppVersion() {
        val version = getString(R.string.version_about, AppGlobal.app_version_name, AppGlobal.app_version_code)
        aboutAppVersion.text = version
    }

    private fun prepareRecyclerView() {
        val manager = LinearLayoutManager(this).apply {
            orientation =
                    if (!resources.getBoolean(R.bool.is_tablet)
                            && resources.configuration.orientation == Configuration.ORIENTATION_LANDSCAPE) RecyclerView.HORIZONTAL
                    else RecyclerView.VERTICAL
        }

        recyclerView.setHasFixedSize(true)
        recyclerView.layoutManager = manager

        refreshLayout.isEnabled = false
    }

    private fun prepareToolbar() {
        toolbar.setNavigationOnClickListener { onBackPressed() }
    }
}